tslearn/docs/integration_other_software.rst

```diff
@@ -0,0 +1,230 @@
+    [array([1., 2.]), array([1., 4., 3.])]
+
+
+sktime
+^^^^^^
+
+`sktime <https://alan-turing-institute.github.io/sktime/>`_ is a ``scikit-learn``
+compatible Python toolbox for learning with time series.
+:func:`tslearn.utils.to_sktime_dataset` and :func:`tslearn.utils.from_sktime_dataset`
+allow users to convert a data set from ``tslearn`` format to ``sktime`` format
+and vice versa.
+``pandas`` is a required dependency to use these functions.
+
+.. code-block:: python
+
+    >>> import pandas as pd
+    >>> from tslearn.utils import from_sktime_dataset, to_sktime_dataset
+    >>> df = pd.DataFrame()
+    >>> df["dim_0"] = [pd.Series([1, 2]), pd.Series([1, 4, 3])]
+    >>> from_sktime_dataset(df)
+    array([[[ 1.],
+            [ 2.],
+            [nan]],
+
+           [[ 1.],
+            [ 4.],
+            [ 3.]]])
+    >>> to_sktime_dataset([[[1], [2], [None]], [[1], [4], [3]]]).shape
+    (2, 1)
+
+
+pyflux
+^^^^^^
+
+`pyflux <https://pyflux.readthedocs.io>`_ is a library for time series analysis
+and prediction.
+:func:`tslearn.utils.to_pyflux_dataset` and :func:`tslearn.utils.from_pyflux_dataset`
+allow users to convert a data set from ``tslearn`` format to ``pyflux`` format
+and vice versa.
+``pandas`` is a required dependency to use these functions.
+
+.. code-block:: python
+
+    >>> import pandas as pd
+    >>> from tslearn.utils import from_pyflux_dataset, to_pyflux_dataset
+    >>> df = pd.DataFrame([1, 2], columns=["dim_0"])
+    >>> from_pyflux_dataset(df)
+    array([[[1.],
+            [2.]]])
+    >>> to_pyflux_dataset([[[1], [2]]]).shape
+    (2, 1)
+
+
+tsfresh
+^^^^^^^
+
+`tsfresh <https://tsfresh.readthedocs.io>`_ iis a python package automatically
+calculating a large number of time series characteristics.
+:func:`tslearn.utils.to_tsfresh_dataset` and :func:`tslearn.utils.from_tsfresh_dataset`
+allow users to convert a data set from ``tslearn`` format to ``tsfresh`` format
+and vice versa.
+``pandas`` is a required dependency to use these functions.
+
+.. code-block:: python
+
+    >>> import pandas as pd
+    >>> from tslearn.utils import from_tsfresh_dataset, to_tsfresh_dataset
+    >>> df = pd.DataFrame([[0, 0, 1.0],
+    ...                    [0, 1, 2.0],
+    ...                    [1, 0, 1.0],
+    ...                    [1, 1, 4.0],
+    ...                    [1, 2, 3.0]], columns=['id', 'time', 'dim_0'])
+    >>> from_tsfresh_dataset(df)
+    array([[[ 1.],
+        [ 2.],
+        [nan]],
+
+       [[ 1.],
+        [ 4.],
+        [ 3.]]])
+    >>> to_tsfresh_dataset([[[1], [2], [None]], [[1], [4], [3]]]).shape
+    (5, 3)
+
+
+cesium
+^^^^^^
+
+`cesium <http://cesium-ml.org>`_ is an open-source platform for time series inference.
+:func:`tslearn.utils.to_cesium_dataset` and :func:`tslearn.utils.from_cesium_dataset`
+allow users to convert a data set from ``tslearn`` format to ``cesium`` format
+and vice versa.
+``cesium`` is a required dependency to use these functions.
```

Comment:
Bcs of this sentence I added the previous comments (Here you explicitly say that the package itself is a dependency, but nowhere else.).

Reply:
Yep, but this is because `cesium` has its own format, hence having the lib installed is required to cast to its format. This does not apply to other libs.

Reply:
Ok loud & clear. Ignore my other comments then!

Reply:
``cesium`` is the only package which is also a dependency to use it because they have their own object to represent time series: `cesium.data_management.TimeSeries`. The other three rely on pandas, the first ones on numpy.